Add tracking support for key HPOS options.

barryhughes opened the pull request:
Track if HPOS is enabled—including relevant details such as if COT or CPT is the authoritative data store, if transactions are in use, etc. Replaces https://github.com/woocommerce/woocommerce/pull/35270 (which tried to do the same with Tracks, instead of WC Tracker).

Closes #34898.

### How to test the changes in this Pull Request:

Using WP CLI, try running:

`wp eval "print_r( WC_Tracker::get_tracking_data()['settings'] );"` 

At the end of the output, you should see a number of HPOS-related options, looking something like this:

```
    [hpos_enabled] => yes
    [hpos_sync_enabled] => yes
    [hpos_cot_authoritative] => yes
    [hpos_transactions_enabled] => no
    [hpos_transactions_level] => REPEATABLE READ
```

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

barryhughes committed:
Add tracing support for key HPOS options.

barryhughes commented:
@peterfabian I hope you don't mind me selecting you directly for the review, felt like a good choice re our earlier conversation.

peterfabian commented on plugins/woocommerce/includes/class-wc-tracker.php:
```diff
@@ -690,6 +690,11 @@ private static function get_all_woocommerce_options_values() {
 			'registration_generate_password'        => get_option( 'woocommerce_registration_generate_password' ),
+			'hpos_enabled'                          => get_option( 'woocommerce_custom_orders_table_enabled' ),
+			'hpos_sync_enabled'                     => get_option( 'woocommerce_custom_orders_table_data_sync_enabled' ),
+			'hpos_cot_authoritative'                => get_option( 'woocommerce_feature_custom_order_tables_enabled' ),
```
I don't think this option determines whether HPOS is authoritative but rather whether the checkbox in Settings > Advanced is checked or not. I'd suggest to rename, and then including the option which says which table is in use.

barryhughes replied:
Yes, they were the wrong way round: thanks for the catch, updated in https://github.com/woocommerce/woocommerce/pull/35446/commits/a15a167d1de3e8b7bf311867133f4ebd103dc4f9.

peterfabian reviewed (changes requested):
I think we need to update the options a bit, please see above.

barryhughes committed:
Fix referenced option keys.

Our usage of the options used to determine if HPOS is enabled and if COT is authoritative were reversed.

peterfabian reviewed (approved):
Looks good now, thanks!
